index.bs

```diff
@@ -3156,7 +3207,15 @@ The following abstract operations support the implementation of the
   1. Set |firstPendingPullInto|'s [=pull-into descriptor/buffer=] to !
      [$TransferArrayBuffer$](|firstPendingPullInto|'s [=pull-into descriptor/buffer=]).
  1. Perform ! [$ReadableByteStreamControllerInvalidateBYOBRequest$](|controller|).
- 1. If ! [$ReadableStreamHasDefaultReader$](|stream|) is true
+ 1. If |controller|.[=ReadableByteStreamController/[[pendingPullIntos]]=] is not
```

Comment:
Can we merge this with the previous "If controller.[[pendingPullIntos]] is not empty" block? I think it should be OK; the intermediate step should not do anything observable...

Reply:
This hinged on what we wanted to do with the errors from `? ReadableByteStreamControllerEnqueueDetachedPullIntoToQueue()`. That is: if that call failed, is it a problem that we did not yet call `ReadableByteStreamControllerInvalidateBYOBRequest()`?

From https://github.com/whatwg/streams/pull/1168#discussion_r779034631, I think the conclusion is: no. If we're going to error the entire stream in response to an allocation failure in `EnqueueDetachedPullIntoToQueue()`, then it doesn't really matter what happens to the BYOB request. So we can do:
1. Transfer `firstPendingPullInto`'s buffer (and throw an error if it's already detached).
2. If `firstPendingPullInto`'s reader type is `"none"`, move its filled bytes to the stream's queue. If this fails, error the stream (and re-throw the error to the caller of `enqueue()`).
3. Invalidate the BYOB request.

Reply:
Hmm, looks like we have the same problem with *remainder* in step 6.2 of [`ReadableByteStreamControllerRespondInReadableState`](https://streams.spec.whatwg.org/commit-snapshots/53282dde15be5f844768c2167405660da3c29b2d/#readable-byte-stream-controller-respond-in-readable-state). If that fails, we also leave the stream in an inconsistent state. 🤔